packages/story-editor/src/components/canvas/editLayerFocusManager/provider.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useReduction, useCallback, useRef } from '@googleforcreators/react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Context from './context';
+import reduction, { INITIAL_STATE } from './reduction';
+import KeyBindings from './keyBindings';
+
+function EditLayerFocusManager({ children }) {
+  const [state, actions] = useReduction(INITIAL_STATE, reduction);
+
+  const stateRef = useRef(state);
+  stateRef.current = state;
```

Comment:
Need both of these?

Reply:
yup! just throwing the state in the ref is nice for functions locally to access it and stay referentially stable downstream. 

I also want to be preemptive around updates here cus the callbacks here are used in layers like the frames layer which can induce some perf costs if not treated carefully